src/mode/modeHandler.ts

```diff
@@ -71,16 +70,31 @@ export class ModeHandler implements vscode.Disposable {
     // This also makes things like gd work.
     // For whatever reason, the editor positions aren't updated until after the
     // stack clears, which is why this setTimeout is necessary
-    setTimeout(() => {
-      this.syncCursors();
-    }, 0);
+    this.syncCursors();
 
     // Handle scenarios where mouse used to change current position.
     const disposable = vscode.window.onDidChangeTextEditorSelection(e => {
       if (configuration.disableExt) {
         return;
       }
 
+      if (Globals.isTesting) {
```

Comment:
Why do you think this is better here instead of in the actual handler?

Do you know that the handler was only called from here?

Reply:
Tbh, I moved it because I didn't see a reason for it to be in the actual handler. This was also the only spot the handler was being called and so decided to move it up here to save us from queuing something that was just going to exit shortly thereafter.

Reply:
ok, we just don't test this function at all so it terrifies me